korma/src/commonMain/kotlin/com/soywiz/korma/interpolation/Easing.kt

```diff
@@ -2,92 +2,171 @@ package com.soywiz.korma.interpolation
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_IN"))
+        val EASE_OUT_IN: Easing get() = Easings.EASE_OUT_IN
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_BACK"))
+        val EASE_IN_BACK: Easing get() = Easings.EASE_IN_BACK
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_BACK"))
+        val EASE_OUT_BACK: Easing get() = Easings.EASE_OUT_BACK
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_OUT_BACK"))
+        val EASE_IN_OUT_BACK: Easing get() = Easings.EASE_IN_OUT_BACK
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_IN_BACK"))
+        val EASE_OUT_IN_BACK: Easing get() = Easings.EASE_OUT_IN_BACK
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_OUT_ELASTIC"))
+        val EASE_IN_OUT_ELASTIC: Easing get() = Easings.EASE_IN_OUT_ELASTIC
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_IN_ELASTIC"))
+        val EASE_OUT_IN_ELASTIC: Easing get() = Easings.EASE_OUT_IN_ELASTIC
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_BOUNCE"))
+        val EASE_IN_BOUNCE: Easing get() = Easings.EASE_IN_BOUNCE
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_OUT_BOUNCE"))
+        val EASE_IN_OUT_BOUNCE: Easing get() = Easings.EASE_IN_OUT_BOUNCE
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_IN_BOUNCE"))
+        val EASE_OUT_IN_BOUNCE: Easing get() = Easings.EASE_OUT_IN_BOUNCE
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_QUAD"))
+        val EASE_IN_QUAD: Easing get() = Easings.EASE_IN_QUAD
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_OUT_QUAD"))
+        val EASE_OUT_QUAD: Easing get() = Easings.EASE_OUT_QUAD
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_IN_OUT_QUAD"))
+        val EASE_IN_OUT_QUAD: Easing get() = Easings.EASE_IN_OUT_QUAD
+        @Deprecated("Moved to Easings enum", ReplaceWith("Easings.EASE_SINE"))
+        val EASE_SINE: Easing get() = Easings.EASE_SINE
     }
 }
 
-private object Easings {
-    private const val BOUNCE_10 = 1.70158
-    private const val PI_2 = PI / 2.0
-
-    val SMOOTH = Easing { it * it * (3 - 2 * it) }
-
-    val EASE_IN_ELASTIC = Easing {
-        if (it == 0.0 || it == 1.0) it else {
-            val p = 0.3
-            val s = p / 4.0
-            val inv = it - 1
-            -1.0 * 2.0.pow(10.0 * inv) * sin((inv - s) * (2.0 * PI) / p)
-        }
-    }
-
-    val EASE_OUT_ELASTIC = Easing {
-        if (it == 0.0 || it == 1.0) it else {
-            val p = 0.3
-            val s = p / 4.0
-            2.0.pow(-10.0 * it) * sin((it - s) * (2.0 * PI) / p) + 1
-        }
-    }
+enum class Easings : Easing {
+    SMOOTH {
+        override fun invoke(it: Double): Double = it * it * (3 - 2 * it)
+    },
```

Comment:
Couldn't be: `SMOOTH(Easing { it * it * (3 - 2 * it) })` ?

Reply:
You mean with enum having a constructor that stores an Easing? I guess, but it seems like (IMO, too much) overhead for just making the syntax more compact.

Reply:
Explained here: https://github.com/korlibs/korge-next/pull/31/files#r491484527 (might require removing the old invoke method)